pandas/core/indexes/multi.py

```diff
@@ -1783,16 +1792,15 @@ def is_lexsorted(self):
     @cache_readonly
     def lexsort_depth(self):
         if self.sortorder is not None:
-            if self.sortorder == 0:
-                return self.nlevels
-            else:
-                return 0
+            return self.sortorder
+
+        return self._lexsort_depth()
```

Comment:
youn't need to make the private function, the point of the cache_readonly decorate is to cache the results here

Reply:
I made the private function to be able to call it in _verify_integrity and prevent duplicate code.
What would you suggest? Does this prevent the result to be cached correctly?

Reply:
ahh i c, ok then. can you add a doc-string to _lexsort_depth in that case (and typing -> int)